### ISSUE TITLE
Add flags to externals to set truth values.

### ISSUE THREAD

**Issue by rkaminsk** [potassco/clingo#124]
```
#external p(X) : q(X). [true]
#external p(X) : q(X). [false]   % default
#external p(X) : q(X). [free]
#external p(X) : q(X). [release] % because available via API
```

@tortinator, @BenKaufmann, @mgebser, @javier-romero: opinions are welcome!

**Comment by tortinator** [comment]
I like your proposal. Do we use 'free' elsewhere as well?  -t

**Comment by rkaminsk** [comment]
In the multi-shot solving with clingo paper we use `unknown` but I don't like that so much. Another option that could work might be `choice`.

**Comment by mgebser** [comment]
I also like the idea. Another possible name for `free` would be `open`. I like these two names most, and they both look likewise good to me.

**Comment by rkaminsk** [comment]
Then I will go for `free` because the same name is used in the API.

**Comment by rkaminsk** [comment]
Implemented in 18e9981993051cfb3216e72797df87c4f8f9324d.
